addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/rekognition/listeners/BaseAsyncResultListener.java

```diff
@@ -0,0 +1,115 @@
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.EventContext;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+public abstract class BaseAsyncResultListener implements PostCommitEventListener {
+
+    private static final Logger log = LogManager.getLogger(BaseAsyncResultListener.class);
+
+    public static final String JOB_ID_CTX_KEY = "jobId";
+
+    @Override
+    public void handleEvent(EventBundle eb) {
+        eb.forEach(this::handleEvent);
+    }
+
+    protected void handleEvent(Event event) {
+        EventContext ctx = event.getContext();
+        if (ctx == null) {
+            return;
+        }
+
+        String jobId = (String) ctx.getProperties().get(JOB_ID_CTX_KEY);
+        if (getFailureEventName().equals(event.getName())) {
+            log.info("Async Rekognition job failed for id " + jobId);
+            return;
+        }
+
+        byte[] bytes = getStore().get(jobId);
+        if (bytes == null) {
+            log.error("Could not find data for JobId " + jobId);
+            return;
+        }
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+             ObjectInputStream ois = new ObjectInputStream(bais)) {
+
+            @SuppressWarnings("unchecked")
+            Map<String, Serializable> params = (Map<String, Serializable>) ois.readObject();
+
+            Collection<EnrichmentMetadata> enrichment = getEnrichmentMetadata(jobId, params);
+            enrichment.forEach(this::saveMetadata);
+        } catch (IOException | ClassNotFoundException e) {
+            log.error(e);
+        }
+    }
+
+    protected void saveMetadata(EnrichmentMetadata metadata) {
+        TransactionHelper.runInTransaction(
+                () -> CoreInstance.doPrivileged(metadata.context.repositoryName, session -> {
+                    DocMetadataService docMetadataService = Framework.getService(DocMetadataService.class);
+                    DocumentModel doc = docMetadataService.saveEnrichment(session, metadata);
+                    if (doc != null) {
+                        try {
+                            session.saveDocument(doc);
+                        } catch (DocumentValidationException e) {
+                            log.warn("Failed to save document enrichment data for {}.", metadata.context.documentRef, e);
+                        }
+                    } else {
+                        log.debug("Failed to save enrichment for document {}.", metadata.context.documentRef);
+                    }
+                })
+        );
+    }
+
+    protected KeyValueStore getStore() {
+        KeyValueService kvs = Framework.getService(KeyValueService.class);
+        return kvs.getKeyValueStore("default");
```

Comment:
EnrichmentService can be confgured per transientStore.  Are you sure you want to hard code it here?

Reply:
don't see why not. we store meta information there

Reply:
using the one from `EnrichmentService` now